engine/src/Branch.java

```diff
@@ -43,4 +48,86 @@ void setHead(Commit newHead) {
     public Commit getStartCommit(){
         return startCommit;
     }
+
+    public void open() {
+        clearCurrentWC();
+        recursiveOpenAllFiles(getRootSha1(), Settings.repositoryFullPath);
+    }
+
+    private void clearCurrentWC() {
+        File directory = new File(Settings.repositoryFullPath);
+        File[] listOfItems = directory.listFiles();
+        for(File item: listOfItems){
+            if(item.isDirectory()){
+                if(!item.getName().equals(Settings.gitFolder)){
+                    deleteSubFilesRec(item);
+                }
+            }
+            else{
+                Utils.deleteFile(item.getPath());
+            }
+        }
+    }
+
+    private void deleteSubFilesRec(File folder) {
```

Comment:
should delete all current repo folder?
if so, use  
```suggestion
    FileUtils.deleteDirectory(destTestsDir);
```
do it all for you in one shot